src/cascadia/TerminalApp/TerminalPage.cpp

```diff
@@ -714,12 +718,15 @@ namespace winrt::TerminalApp::implementation
     //   TitleChanged event.
     // Arguments:
     // - tab: the Tab to update the title for.
-    void TerminalPage::_UpdateTitle(std::shared_ptr<Tab> tab)
+    void TerminalPage::_UpdateTitle(winrt::TerminalApp::Tab tab)
     {
-        auto newTabTitle = tab->GetActiveTitle();
+        winrt::com_ptr<Tab> newTabImpl;
+        newTabImpl.copy_from(winrt::get_self<implementation::Tab>(tab));
```

Comment:
you may(?) be able to oneshot this:

```c++
auto newTabImpl{ winrt::get_self<implementation::Tab>(tab) };
```

Reply:
It seems `get_self` would return a `implementation::Tab*`, but I remember reading the docs, where it said I should be putting the result in a `com_ptr` using `copy_from` so that the original object doesn't disappear from beneath my legs, which is why I have these wonky two lines.